docs/spec.md

```diff
@@ -104,6 +107,15 @@ type with shape `2x3` and element type `f32`. It has two dimensions (or,
 in other words, two axes) - 0th dimension and 1st dimension - whose sizes are
 2 and 3. Its rank is 2.
 
+This defines support for **static shapes** where dimension sizes are statically
+known. In the future, we are planning to also introduce support for
+**dynamic shapes** where dimension sizes are either partially or fully unknown
+([#8](https://github.com/openxla/stablehlo/issues/8)). Furthermore, we are
+planning to explore extending tensor types beyond dimension sizes and element
+types, for example, to include layouts
+([#629](https://github.com/openxla/stablehlo/issues/629)) and sparsity
+([#1078](https://github.com/openxla/stablehlo/issues/1078)).
```

Comment:
Good catch! As I was working on this pull request, I also called out several tickets which are not part of the Frontend Contract project, because they represented highly likely todos from the New Features project. I think this is also useful to include in the spec for the readers to understand the anticipated direction of StableHLO evolution. I've called this our in the PR description.